Sources/Core/SoftwareBreakpointManager.cpp

```diff
@@ -124,4 +124,31 @@ bool SoftwareBreakpointManager::fillStopInfo(Target::Thread *thread,
   stopInfo.reason = StopInfo::kReasonBreakpoint;
   return true;
 }
+
+void SoftwareBreakpointManager::insertStashedInsns(Address const &start,
+                                                   size_t length,
+                                                   ByteVector &data) {
+  Address end = start + length;
+  for (auto const &insn : _insns) {
+    Address insn_start = insn.first;
+    Address insn_end = insn.first + insn.second.size();
+
+    // no overlap between region and instruction
+    if (end < insn_start || start >= insn_end) {
+      continue;
+    }
+
+    // overlap begins at beginning of instruction
+    if (start <= insn_start) {
+      uint64_t offset = insn_start - start;
+      uint64_t size = std::min((uint64_t)insn.second.size(), end - insn_start);
+      memcpy(&data[offset], &insn.second[0], size);
+    } else {
+      DS2ASSERT(start < insn_end);
+      uint64_t offset = insn_end - start;
+      uint64_t size = start - insn_end;
```

Comment:
It should be the case that `start` is strictly less than `insn_end`, as shown by the assert and continue above. This result will be negative, which I'm sure is not what you wanted.

I think the correct thing to do here is `insn_end - start`, since that should be the number of bytes you want to write in this case.